src/main/plugins/org.dita.pdf2/xsl/fo/tables.xsl

```diff
@@ -216,6 +216,19 @@ See the accompanying LICENSE file for applicable license.
                                                     '+-0123456789.'))"/>
   </xsl:function>
   
+  <xsl:function name="dita-ot:get-entry-end-position" as="xs:integer">
+    <xsl:param name="thisentry" as="element()"/>
+    <xsl:choose>
+      <xsl:when test="$thisentry/@dita-ot:x and $thisentry/@dita-ot:morecols">
+        <xsl:sequence select="xs:integer($thisentry/@dita-ot:x) + xs:integer($thisentry/@dita-ot:morecols)"/>
+      </xsl:when>
+      <xsl:when test="$thisentry/@dita-ot:x">
+        <xsl:sequence select="xs:integer($thisentry/@dita-ot:x)"/>
+      </xsl:when>
+      <xsl:otherwise>0</xsl:otherwise>
```

Comment:
Saxon probably doesn't care, but I would prefer to use

```xml
<xsl:sequence select="0"/>
```

To make it easier to read that we're returning an integer instead of text node with content "0".